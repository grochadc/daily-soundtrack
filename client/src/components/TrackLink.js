import React from "react";
import { Link } from "@reach/router";
import { Panel, Media } from "react-bootstrap";
import { distanceInWordsToNow } from "date-fns";
import "../index.css";

function TrackLink(props) {
  let { track_info, message } = props;

  return (
    <Panel>
      <Panel.Body>
        <Link to={"/player/" + props._id} style={{ textDecoration: "none" }}>
          <Media>
            <Media.Left>
              <img alt={""} src={track_info.art} width="100" height="100" />
            </Media.Left>
            <Media.Body>
              <h3>{track_info.title}</h3>
              by {track_info.artist}
              <p>{message}</p>
            </Media.Body>
          </Media>
        </Link>
      </Panel.Body>
      <Panel.Footer>
        Posted by: <Link to={"/playlist/" + props.user}>{props.user}</Link>{" "}
        <div className="small">
          {distanceInWordsToNow(new Date(props.date))} ago
        </div>
        <br />
      </Panel.Footer>
    </Panel>
  );
}

export default TrackLink;
