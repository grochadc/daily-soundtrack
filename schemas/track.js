const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const trackSchema = new Schema({
  track_info: {
    title: String,
    artist: String,
    album: String,
    art: String
  },
  uri: String,
  date: Date,
  user: {
    id: String,
    name: String
  },
  message: String
});

module.exports = trackSchema;
