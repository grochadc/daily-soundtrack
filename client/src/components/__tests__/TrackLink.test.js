import React from "react";
import renderer from "react-test-renderer";
import track from "./trackpost.json";
import TrackLink from "../TrackLink";

jest.mock("react-bootstrap");

test("Renders a track link", () => {
  const component = renderer.create(<TrackLink {...track} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
