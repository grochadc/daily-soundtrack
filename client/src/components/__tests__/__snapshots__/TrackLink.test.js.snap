// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a track link 1`] = `null`;
